### ISSUE TITLE
Virtual screen reader interactivity?

### ISSUE THREAD

**Issue by DarrenXu94** [guidepup/virtual-screen-reader#44]
Hi there,

Great initiative and great library! I am trying to use virtual-screen-reader with my Vue components and testing-library. Is it possible to have events from JS or interactive elements captured? I've noticed `fireEvent` from `'@testing-library/vue'` doesn't change anything and neither does `virtual.press('Enter')`. Not sure if it's possible but I don't see any examples.

Thanks!

**Comment by jlp-craigmorten** [comment]
Hi @DarrenXu94 👋 

Please see https://github.com/guidepup/virtual-screen-reader/blob/main/examples/vue/src/__tests__/IncrementCounter.spec.ts as an example of writing a Virtual Screen Reader test with an interactive Vue component.

As far as I could tell everything was working as expected.

If you have specific issues, or an example that isn't working, happy to take a look and potentially reopen the issue.

**Comment by DarrenXu94** [comment]
Hi there, I have created a playground with the component to test [here](https://play.vuejs.org/#eNqtVuFu4kYQfpWpVQmQgBCd+scNaXOU6qguSZXj2j+WTht7gL1b72531wREePfOro1xDDlFalEU8M7MNzPfjGd2F91oPVwXGMXRlcNcC+bwOpEA9LRxzCCDVDBrx0m0UCYfpEo6o0QSwa9c6sLR+YrJTODMP9FxnLONQLl0KxLVv5Po+urigFjiW81kjZ2umBkInnMHXmuQMu24kmS224HBnHHJ5XJCWix1aCzs9+BtqieBC3d14RHPYK+5LZgQ28GKZxkSJjDDGXlbI0m1EtxhEnk7gFfc/a7MjZSqkCnmKN2r3j3IWyLwbmpuWmikqJ4wgzofT1xdGHq0qeHagWBySdjOElwiea6VcbCDDBdc4kTRswyhwsKoHDpU4443pwJaB79N399/vptMv0w+3Dx8+Ti7nc1hDJcjuLiAv1cogQpRRcd9gtaCozrTP277wCouANdotvANt9qQygn4fHY7vf9cIo8C9qWlAB9LY+qnGsmzSvaJxE3Ig7JghQjfzWy6O0+uZDnG0JlX7TQhBGKu0/cybZS2MQQ9OHJcnwC4rSbruyJ/RBNM/KfyF8NPo/JoX4lylaH4i4mCbFoQn5yhJjmF6JSR+I/BfwpuMIthwYTFCtp/lfgZc6zbOwAbdIWRRzcHpuY8RzMjkBEwCzJEDs9QyJKdrOHupHPjULLhsdkG0O2WR3VmHtWGZJqovV+GlcnzM4x633PSej3igyox9f84PyEuR7dS2bHQjRnUpdJM1xRGL4a14tmRz7I718HrGLo4pPZZovMRfJjffvT9dEP9NBUhCwohqNZF93H/iDSiup1CU+UwPmbR6Ze4FOkL/XOja3zKSrCtMn5hX+b1cIrS5LvbND9wVdXrbQBkWnchAF9AVaVGiPQDrsbnBkfDkjgWyIxvWEWVCCCtLm4wVHd8qfiGuXuGOh/6ATA0yFujaLDcEpIbi+5gzuxWpkBv6fi6meh/C7mOuH8yLV80eyL39EyrIMx8vwys2wq/CIatpVLGppXlfnHGwB6tEoVD+KFcDUy6n73GE8/8OLzUm7ZkhXy5ovl1RqRZllGWfgK1JDm9QpzcDc5YKdoOC1pmMVQhtuSp4DqmHkhdd9SH6q93EvCK9vOAdmFKL7ZUT4bptsqjMhmak+iIPWKu4ivq06qkCbDgy+FXqyRdeAJjdPeg7cIFmvtw5aB1Wk+VJArL+I9w5kyB1Uvl7yuYfjtz/tVu/FkS/UkLEc2aLha1rJw2pXj66Y6mTUNIo6QQpP0d4QOGioZrkVd7T9OSwm7ohWhngQGq1txONw6lPSTlAz1OULqVFGGxvpb6Mdx3w3eHZoz2/wKI/3SV)

I am running this test against it
```
import { describe, it } from 'vitest'
import { render, screen, waitFor } from '@testing-library/vue'
import TextareaCounter from '../TextareaCounter.vue'

import { virtual } from '@guidepup/virtual-screen-reader'


describe('Textarea accessibility', () => {
  it('announces chars remaining', async () => {
    const { container } = render(TextareaCounter)

    virtual.start({ container })

    // Checking we are within textarea
    console.log(virtual.activeNode)

    await virtual.type('test')
   
    await waitFor(() => screen.getByDisplayValue('test'))

    console.log(await virtual.spokenPhraseLog());
    await virtual.stop();  
  })
})
```

Basically what should happen is it debounces the announcement for chars remaining.

When using NVDA I get the result 
<img width="263" alt="Screenshot 2024-01-31 134732" src="https://github.com/guidepup/virtual-screen-reader/assets/16931153/89bcd291-be4e-432b-b07b-377a56e14c4f">


When I log out the spokenPhraseLog it just returns `[ 'textbox' ]`

Any ideas?

**Comment by jlp-craigmorten** [comment]
Hi @DarrenXu94 

I've mirrored your example with working tests as a new example, see https://github.com/guidepup/virtual-screen-reader/blob/main/examples/vue/src/__tests__/TextAreaCounter.spec.ts

You test logic was flawed in that it doesn't take into account the 1 second delay of the debounce you have put in place - you are getting the spoken phrase log _before_ the time ellapses to actually update your hidden live region.

**Comment by DarrenXu94** [comment]
Thanks so much for all your help! One final thing, I copied the component and test locally, ran the test and got this error
<img width="962" alt="image" src="https://github.com/guidepup/virtual-screen-reader/assets/16931153/76a6e14f-63bc-4d8d-afb1-61e8fe1cd424">

Seems to be a different output from the expected snapshot.

This is not stopping me or anything, I'd just like to let you know the test was failing, not sure if there are different results based on different systems (I'm using Node v18.19.0 on Windows) or if the test didn't work in the first place

Thanks again

**Comment by cmorten** [comment]
Ah I believe that is a small bug that was fixed in `0.16.2` where element being typed into was incorrectly being announced again after live region announcements.

If you upgrade to the latest the snapshots should hopefully match the examples.

**Comment by DarrenXu94** [comment]
Hi there, sorry another issue with a slightly more complex example
[here](https://play.vuejs.org/#eNq1VUuP2zYQ/iuEWsD2Yi27TU+qd9Gm3UMLtFs0OepCSSOJuxSpkJRXzsL/PcOHaHkfARIk9sHDeXE4833jx+T3vk/3AyRZsjPQ9ZwauM4FIbuK7Z2AYjEYIwX5reSsvL/KE6ZvexDkihg1QJ5cu1MnK8p3G+87Rf5jlSST6BDj8sRmkhrmmWrKtU3lA0opDAiz27ijr2fjC9ptZmXiUZeK9YZwKhrMZzTmyAXreqkMeSQV1EzAHxLPAhOSI6mV7MgCX7yIbv5Ob0j9lbYl6JALGJ0L5qEDd7/zfMtHW5qgHWRk4QJvRorVweLSGsrJT2fEeRJ/lxWPzqOihi5Xk1GBGZSYToT47mShOU55dLG5OK5sK/zj8cHJJT4du1azJr3TUuA4XZY8sTUwDuq2N0wKbE8sJU8o5/Lhb6ezk3QVuZgWyvsX9Hd6tLo8+U+BBrXHgUWboaoB48037/6FEeVoRGwMHL0/Y/wftOSDrdG7vR1EhWXP/Fy1f7mRMdG81zcjYkRPj3JQjH3NExygHdNrTz+V+yZ9M/UUuxin/5wQt3tQnB5Iplv5MEPzhHXEJ1GSW1hXjHLZINBLTrVGhSPHOsAa9VQxuua0AI5Gjz/DjG1RyIb5NJfmGmdsf6LyKRV/hI6Z5cLxabHCzKxCLYzMrL0rZrzB01NiBjZZKbzr+3BratrErnD+an69SqyQ+ASBB2rKNprtBooHQlrq0LWcGFZIHBwVq4zsJatOfoSYluk0rqkwc/xOdlaTkCXS2H80mPesAzmYJTL86vrcat8gtEHKa0xbyXLo8DnphwHU4R1wKI1Uy8UPs0HidM/i7cUYvbIp0hoT6OWZh90QUZ7EIPgt4trUK9mfuri5uAiuF24JI541tgCIxfymZRVKBvFBZO3UDtdTxObFVptDj9N76xscyGxX3YeBKajCcov6AIQXdt433Zc7bQ7cojw9Y6ZPVdDyvlESFxDquVQZaRQcfnX3M438OOAVHEanoZw1Ys2QONjFEnOAcvq7QRtWH6bMZ7aOiXULrGlR/dN2u29RizVifaEqXEQzojxfRbqnguzXrEZa2smclpBB7DhGGYm2QlaHs5Vi46adJANjTvawdMjmFLGxIXFpTNnD5gi2b7k2vmgjhB75nfA6kP8EbDz2HKmGpAnPtogeeEUKj22RPkGxVX4Bimf/k58D8TMgzqE4VeYujUgruCzvHWx6qZn9J8O8bITK6YzsM7J1Iocar/Sy8uDyh0Li/ujiKaI7I6op6LLgqLkk2/SXlXP4uGaighGh+fPWhpwj8/gJtkBURQ==)

In this case I'm opening a modal using tab and enter keys.  The modal has the role dialog and focuses on the exit button when it's open. 

<img width="203" alt="Screenshot 2024-02-07 085427" src="https://github.com/guidepup/virtual-screen-reader/assets/16931153/eb409069-2570-4c6a-85cd-c8d2b72399f1">

The expected behaviour is when the focus is within a dialog it announces 'dialog' and the title.

This is the test I'm running against it


```
import { afterEach, describe, expect, it } from 'vitest'
import { render, screen, waitFor } from '@testing-library/vue'
import ModalExample from '../ModalExample.vue'

/**
 * Replace with:
 *
 * import { virtual } from '@guidepup/virtual-screen-reader'
 *
 * in your own code.
 */
import { virtual } from '@guidepup/virtual-screen-reader'

describe('Text Area Counter', () => {
  afterEach(async () => {
    await virtual.stop()
  })

  it('announces the modal as a dialog', async () => {
    const { container } = render(ModalExample)

    await virtual.start({ container: document.body })
    await virtual.next()
    await virtual.act()

    // Adding timeout to counteract the setTimeout in Modal.vue
    await new Promise((resolve) => setTimeout(resolve, 1000))

    await waitFor(() =>screen.getByText('Modal content'))

    console.log(await virtual.spokenPhraseLog())
  })  
})
```

and the result is `[ 'document', 'button, Open modal', 'button, Exit' ]`

so I see the modal is open and the button has focus because 'button, Exit' is announced, but no announcement for the dialog.

Sorry for being annoying and asking so many questions but I am very excited for using this library

**Comment by jlp-craigmorten** [comment]
Hey @DarrenXu94 

For this particular example see #12 where haven't completed all the modal behaviours yet, will take a look at it.

**Comment by jlp-craigmorten** [comment]
Ok, see `0.17.0` which should have the feature along with the example https://github.com/guidepup/virtual-screen-reader/blob/main/examples/vue/src/__tests__/OpenModal.spec.ts demonstrating the dialog being announced when it is entered.
